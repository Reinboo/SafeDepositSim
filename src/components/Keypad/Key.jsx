import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Key = ({ keyFace, handleClick }) => (
  <Wrapper type="button" id={keyFace} onClick={handleClick}>
    {keyFace}
  </Wrapper>
);

Key.propTypes = {
  keyFace: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

const Wrapper = styled.button`
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ${props => props.theme.buttonColor};
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ${props => props.theme.fontMedium};
  color: ${props => props.theme.buttonTextColor};
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ${props => props.theme.buttonColorPressed};
  }
`;

export default Key;
