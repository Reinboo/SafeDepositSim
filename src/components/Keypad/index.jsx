import React from 'react';
import styled from 'styled-components';

import Key from './Key';

const keyFaces = ['7', '8', '9', '4', '5', '6', '1', '2', '3', '*', '0', 'L'];

const Keypad = () => (
  <Wrapper>
    {keyFaces.map(value => (
      <Key keyFace={value} />
    ))}
  </Wrapper>
);

const Wrapper = styled.div`
  margin: 1rem 0;

  display: grid;
  grid-template-rows: repeat(4, 1fr);
  grid-template-columns: repeat(3, 1fr);
`;

export default Keypad;
