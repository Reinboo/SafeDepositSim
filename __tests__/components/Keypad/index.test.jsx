import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Keypad from '../../../src/components/Keypad';
import Key from '../../../src/components/Keypad/Key';

describe('Keypad component', () => {
  test('renders correctly', () => {
    const mockHandleLock = jest.fn();
    const mockHandleUpdate = jest.fn();
    const wrapper = mount(
      <Keypad
        handleLock={mockHandleLock}
        handleUpdate={mockHandleUpdate}
        isServiceMode={false}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('supplies proper handler on L press', () => {
    const mockHandleLock = jest.fn();
    const mockHandleUpdate = jest.fn();
    const wrapper = mount(
      <Keypad
        handleLock={mockHandleLock}
        handleUpdate={mockHandleUpdate}
        isServiceMode={false}
      />,
    );
  });


});
