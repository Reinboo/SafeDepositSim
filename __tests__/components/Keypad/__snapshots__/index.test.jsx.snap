// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad component renders correctly 1`] = `
<Keypad
  handleLock={[MockFunction]}
  handleUpdate={[MockFunction]}
  isServiceMode={false}
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "ggxcE",
            "rules": Array [
              "
  margin: 1rem 0;

  display: grid;
  grid-template-rows: repeat(4, 1fr);
  grid-template-columns: repeat(3, 1fr);
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bwzfXH ggxcE"
      >
        <Key
          handleClick={[MockFunction]}
          key="7"
          keyFace="7"
        >
          <styled.button
            id="7"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="7"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="7"
                onClick={[MockFunction]}
                type="button"
              >
                7
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="8"
          keyFace="8"
        >
          <styled.button
            id="8"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="8"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="8"
                onClick={[MockFunction]}
                type="button"
              >
                8
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="9"
          keyFace="9"
        >
          <styled.button
            id="9"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="9"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="9"
                onClick={[MockFunction]}
                type="button"
              >
                9
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="4"
          keyFace="4"
        >
          <styled.button
            id="4"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="4"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="4"
                onClick={[MockFunction]}
                type="button"
              >
                4
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="5"
          keyFace="5"
        >
          <styled.button
            id="5"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="5"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="5"
                onClick={[MockFunction]}
                type="button"
              >
                5
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="6"
          keyFace="6"
        >
          <styled.button
            id="6"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="6"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="6"
                onClick={[MockFunction]}
                type="button"
              >
                6
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="1"
          keyFace="1"
        >
          <styled.button
            id="1"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="1"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="1"
                onClick={[MockFunction]}
                type="button"
              >
                1
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="2"
          keyFace="2"
        >
          <styled.button
            id="2"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="2"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="2"
                onClick={[MockFunction]}
                type="button"
              >
                2
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="3"
          keyFace="3"
        >
          <styled.button
            id="3"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="3"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="3"
                onClick={[MockFunction]}
                type="button"
              >
                3
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="*"
          keyFace="*"
        >
          <styled.button
            id="*"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="*"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="*"
                onClick={[MockFunction]}
                type="button"
              >
                *
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="0"
          keyFace="0"
        >
          <styled.button
            id="0"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="0"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="0"
                onClick={[MockFunction]}
                type="button"
              >
                0
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
        <Key
          handleClick={[MockFunction]}
          key="L"
          keyFace="L"
        >
          <styled.button
            id="L"
            onClick={[MockFunction]}
            type="button"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "cponZC",
                    "rules": Array [
                      "
  width: 4rem; height: 4rem;
  margin: .5rem;
  border: 1px solid rgba(0, 0, 0, .5);
  border-radius: 5px;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: ",
                      [Function],
                      ";
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, .5);
  cursor: pointer;
  font: 2.5rem ",
                      [Function],
                      ";
  color: ",
                      [Function],
                      ";
  
  transition: background-color .2s linear;
  
  &:hover {
    background-color: ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="L"
              onClick={[MockFunction]}
              type="button"
            >
              <button
                className="sc-bdVaJa cponZC"
                id="L"
                onClick={[MockFunction]}
                type="button"
              >
                L
              </button>
            </StyledComponent>
          </styled.button>
        </Key>
      </div>
    </StyledComponent>
  </styled.div>
</Keypad>
`;
